src/Lecture2.hs

```diff
@@ -48,7 +51,13 @@ zero, you can stop calculating product and return 0 immediately.
 84
 -}
 lazyProduct :: [Int] -> Int
-lazyProduct = error "TODO"
+lazyProduct [] = 1
+lazyProduct list = go 1 list
+  where
+    go :: Int -> [Int] -> Int
+    go 0 _             = 0
+    go result []       = result
+    go result (x : xs) = go (result * x) xs
```

Comment:
That's a very clever solution! 🧠 
I'm not sure if you know about BangPatterns or strict evaluations in Haskell (I'm going to cover them tomorrow in Lecture 3) but your solution also has the optimal performance behaviour 😅

Reply:
Thanks :-) In the past, I have mostly been reading about Haskell, haven't written any code. I have come across BangPatterns and strict evaluation in my readings, but never applied them as I've not actually written any Haskell code.

Reply:
by the way @chshersh curious, how do you get these cool emojis in your comments?

Reply:
@kpadmasola You can start typing : and then the name of an emoji. GitHub has some autocompletion for emoji names 🙂 Like `:smile` or `:+1`.